src/lib/bitenc.rs

```diff
@@ -0,0 +1,571 @@
+    /// // Add another 2 to create a second block
+    /// bitenc.push(2);
+    /// assert_eq!(bitenc.len(), 5);
+    /// ```
+    #[deprecated(
+        since = "0.33.0",
+        note = "Please use the more specific `nr_blocks` and `nr_symbols` functions instead."
+    )]
+    pub fn len(&self) -> usize {
+        self.len
+    }
+
+    /// Get the number of blocks used by the encoding.
+    ///
+    /// Complexity: O(1)
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use fqtk_lib::bitenc::BitEnc;
+    ///
+    /// let mut bitenc = BitEnc::new(8);
+    /// bitenc.push(2);
+    /// assert_eq!(bitenc.nr_blocks(), 1);
+    /// // Add enough 2s to completely fill the first block
+    /// bitenc.push(2);
+    /// bitenc.push(2);
+    /// bitenc.push(2);
+    /// assert_eq!(bitenc.nr_blocks(), 1);
+    /// // Add another 2 to create a second block
+    /// bitenc.push(2);
+    /// assert_eq!(bitenc.nr_blocks(), 2);
+    /// ```
+    pub fn nr_blocks(&self) -> usize {
+        self.storage.len()
+    }
+
+    /// Get the number of symbols encoded.
+    ///
+    /// Complexity: O(1)
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use fqtk_lib::bitenc::BitEnc;
+    ///
+    /// let mut bitenc = BitEnc::new(8);
+    /// bitenc.push(2);
+    /// assert_eq!(bitenc.nr_symbols(), 1);
+    /// bitenc.push(2);
+    /// bitenc.push(2);
+    /// bitenc.push(2);
+    /// assert_eq!(bitenc.nr_symbols(), 4);
+    /// bitenc.push(2);
+    /// assert_eq!(bitenc.nr_symbols(), 5);
+    /// ```
+    pub fn nr_symbols(&self) -> usize {
+        self.len
+    }
+
+    /// Is the encoded sequence empty?
+    ///
+    /// Complexity: O(1)
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use fqtk_lib::bitenc::BitEnc;
+    ///
+    /// let mut bitenc = BitEnc::new(2);
+    /// assert!(bitenc.is_empty());
+    /// bitenc.push(2);
+    /// assert!(!bitenc.is_empty());
+    /// bitenc.clear();
+    /// assert!(bitenc.is_empty());
+    /// ```
+    pub fn is_empty(&self) -> bool {
+        self.len == 0
+    }
+
+    /// Calculate the Hamming distance between this and another bitencoded sequence.
+    ///
+    /// Note: this allows IUPAC fuzzy matching with IUPAC bases in this sequence.
+    /// An IUPAC base in the other sequence matches if it is at least as specific as the
+    /// corresponding (IUPAC) base in this sequence. E.g. If the other sequence is an
+    /// N, it will not match anything but an N, and if the other base is an R, it
+    /// will match R, V, D, and N, since the latter IUPAC codes allow both A and G.
+    pub fn hamming(&self, other: &BitEnc, max_mismatches: u32) -> u32 {
+        assert!(self.len == other.len, "Both bitenc sequences must have the same length");
+        assert!(self.width == other.width, "Both bitenc sequences must have the same width");
+        let mut count: u32 = 0;
+        let values_per_block = self.usable_bits_per_block / self.width;
+        for block_index in 0..self.nr_blocks() {
```

Comment:
Nit: this could be written using `(0..self.nr_blocks()).into_iter().scan(..)` to avoid the need for `let mut`.